Dockerfile

```diff
@@ -0,0 +1,68 @@
+# This is the layer that can run things
+FROM debian:buster
+LABEL maintainer="Binary.com <binary@cpan.org>"
+
+# Some standard server-like config used everywhere
+ENV TZ=UTC
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Use an apt-cacher-ng or similar proxy when available during builds
+ARG DEBIAN_PROXY
+
+WORKDIR /usr/src/perl
+
+RUN [ -n "$DEBIAN_PROXY" ] \
+ && (echo "Acquire::http::Proxy \"http://$DEBIAN_PROXY\";" > /etc/apt/apt.conf.d/30proxy) \
+ && (echo "Acquire::http::Proxy::ppa.launchpad.net DIRECT;" >> /etc/apt/apt.conf.d/30proxy) \
+ || echo "No local Debian proxy configured" \
+ && apt-get update \
+ && apt-get dist-upgrade -y -q --no-install-recommends \
+ && apt-get install -y -q --no-install-recommends \
+    git openssh-client curl socat ca-certificates gcc make libc6-dev libssl-dev zlib1g-dev xz-utils dumb-init \
+ && curl -SL https://www.cpan.org/src/5.0/perl-5.32.0.tar.xz -o perl-5.32.0.tar.xz \
+ && echo '6f436b447cf56d22464f980fac1916e707a040e96d52172984c5d184c09b859b *perl-5.32.0.tar.xz' | sha256sum -c - \
+ && tar --strip-components=1 -xaf perl-5.32.0.tar.xz -C /usr/src/perl \
+ && rm perl-5.32.0.tar.xz \
+ && ./Configure -Duse64bitall -Duseshrplib -Dprefix=/opt/perl-5.32.0 -Dman1dir=none -Dman3dir=none -des \
+ && make -j$(nproc) \
+ && make install \
+ && cd /usr/src \
+ && curl -LO https://www.cpan.org/authors/id/M/MI/MIYAGAWA/App-cpanminus-1.7044.tar.gz \
+ && echo '9b60767fe40752ef7a9d3f13f19060a63389a5c23acc3e9827e19b75500f81f3 *App-cpanminus-1.7044.tar.gz' | sha256sum -c - \
+ && tar -xzf App-cpanminus-1.7044.tar.gz \
+ && rm App-cpanminus-1.7044.tar.gz \
+ && cd App-cpanminus-1.7044 && /opt/perl-5.32.0/bin/perl bin/cpanm . \
+ && rm -rf /var/lib/apt/lists/* /var/cache/apt/* \
+ && rm -fr ./cpanm /root/.cpanm /usr/src/perl /usr/src/App-cpanminus-1.7044* /tmp/* \
+# Locale support is probably quite useful in some cases, but
```

Comment:
Still unsure whether we should keep this - clean Unicode handling is an important feature, so maybe we *should* enforce a `.UTF-8` locale by default...?

Reply:
Maybe, though with the current `debian:buster` base we already fall back to a `C.UTF_8` locale:

```
root@558bed5a8132:/app# locale -a
C
C.UTF-8
POSIX
root@558bed5a8132:/app# ls /usr/lib/locale/
C.UTF-8
root@558bed5a8132:/app# locale
LANG=
LANGUAGE=
LC_ALL=
```